utils/R/convert.input.R

```diff
@@ -22,73 +23,193 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
+                                                             host, user = NA, verbose = TRUE,R = "R")
+                        
+                        file.deletion.commands <- .get.file.deletion.commands(files.to.delete)
+                        
+                        remote.execute.R( file.deletion.commands$move.to.tmp,
+                                          host, user = NA, 
+                                          verbose = TRUE,R = "R")
+                        
+                        # Schedule files to be replaced or deleted on exiting the function
+                        successful <- FALSE
+                        on.exit(if (exists("successful") && successful) {
+                          
+                          logger.info("Conversion successful, with overwrite=TRUE. Deleting old files.")
+                          remote.execute.R( file.deletion.commands$delete.tmp, host, user = NA, verbose = TRUE,  R = "R" )
+                          
+                        } else {
+                          logger.info("Conversion failed. Replacing old files.")
+                          remote.execute.R(file.deletion.commands$replace.from.tmp, host, user = NA, verbose = TRUE, R = "R" )
+                        })
+          }
+      
+    
+
+      # There's an existing input that matches desired start/end dates. Use that one.
       logger.info("Skipping this input conversion because files are already available.")
       return(list(input.id = existing.input$id, dbfile.id = existing.dbfile$id))
-    } else {
-      # Start/end dates need to be updated so that the input spans a continuous
-      # timeframe
-      start_date <- min(start_date, existing.input$start_date)
-      end_date <- max(end_date, existing.input$end_date)
-      logger.info(paste0("Changed start/end dates to '", 
-                         start_date, "'/'", end_date, "' ", 
-                         "so that existing input can be updated while maintaining continuous time span."))
       
-      # There might be existing files for some years (but not all; checked that above)
-      # fcn should be smart enough not overwrite the existing ones, and hopefully won't
-      # waste time working on them either At the end, if convert.inputs was successful
-      # we'll need to update its start/end dates We don't know the dbfile path/prefix
-      # until after fcn runs, so unfortunately can't check that the new dbfile record
-      # won't conflict with existing ones.
-    }
-  } else {
-    # No existing record found. Should be good to go.
-  }
+      
+      } else{
+      # No existing record found. Should be good to go with regular conversion.
+     }
+    
+    ##-------------------------end of exact.dates chunk------------------------------------#
+  
+    }else{
+    
+      ## Either Running met2model for models that do not need exact dates
+      ## Or running non met2model conversion scripts (download.raw, met2cf, etc.)
```

Comment:
This comment is misleading, as convert.inputs can (and hopefully will) be used outside of met.process.  You are simply running conversions that do not need exact date matches